app/lib/study_search_service.rb

```diff
@@ -55,4 +55,124 @@ def self.sanitize_gene_params(genes)
     end
     gene_array.map(&:strip)
   end
+
+  # generate a Mongoid::Criteria object to perform a keyword/exact phrase search based on contextual use case
+  # supports the following query_contexts: :keyword (individual terms), :phrase (quoted phrases & keywords)
+  # and :inferred (converting a facet-based query to keywords)
+  # will scope the query based off of :base_studies, and include/exclude studies matching
+  # :accessions based on the :query_context (included by default, but excluded in :inferred to avoid duplicates)
+  def self.generate_mongo_query_by_context(terms:, base_studies:, accessions:, query_context:)
+    case query_context
+    when :keyword
+      author_match_study_ids = Author.where(:$text => { :$search => terms }).pluck(:study_id)
+
+      matches_by_text = base_studies.where({ :$text => { :$search => terms } })
+      metadata_matches = get_studies_from_term_conversion(terms.split)
+      all_accessions = accessions + metadata_matches.keys
+      matches_by_accession = base_studies.where({ :accession.in => all_accessions })
```

Comment:
`matches_by_accession` looks like it's over-reporting the things that have matched by accession, or at least something is funny with the math I'm seeing - see attached screenshots for where `matches_by_accession` has the same number as the total results but other match properties also claim results. 

<img width="1274" alt="Screen Shot 2022-04-25 at 4 06 37 PM" src="https://user-images.githubusercontent.com/54322292/165167204-ba6cf89c-e8b7-4616-9f38-fa7b19598bc8.png">
<img width="1296" alt="Screen Shot 2022-04-25 at 4 10 10 PM" src="https://user-images.githubusercontent.com/54322292/165167225-daf30fa2-bebe-44c4-b160-b58905f19e36.png">

It is not happening when results are purely text or author based
<img width="1208" alt="Screen Shot 2022-04-25 at 4 14 39 PM" src="https://user-images.githubusercontent.com/54322292/165167682-25820093-d967-49ab-b764-fde8ad8aca8d.png">
<img width="1309" alt="Screen Shot 2022-04-25 at 4 15 02 PM" src="https://user-images.githubusercontent.com/54322292/165167686-274d20b9-78a9-4b0b-9ec2-093f162b6323.png">

Reply:
@ehanna4 you are correct - it was.  With Eric's suggestion of adding the `numResults:scp:metadata` property, this is now resolved.  The underlying query still finds the studies via their accession (after doing the metadata match), but we are able to report them discretely.  Now, only studies queried for using their accessions (SCPXXXX) will roll up on `numResults:scp:accession`.